packages/grafana-ui/src/components/Combobox/Combobox.tsx

```diff
@@ -156,6 +156,12 @@ export const Combobox = <T extends string | number>(props: ComboboxProps<T>) =>
     [createCustomValue, id, ariaLabelledBy]
   );
 
+  // Memoize for using in fuzzy search
+  const stringifiedItems = useMemo(
+    () => (isAsync ? [] : options.map((item) => itemToString(item))),
+    [options, isAsync]
+  );
```

Comment:
Lets keep an eye on this/where we call `itemToString`. If we're running it over all the items here, it would be good to reuse these calls elsewhere. I don't think we need to fix it immediately, but someone to look at in the future.

Reply:
`itemToString ` was only used in the original filtering, so for now I think we're okay. Good catch though.